eureka2-testkit/src/main/java/com/netflix/eureka2/testkit/embedded/server/EmbeddedReadServer.java

```diff
@@ -80,21 +81,21 @@ public void start() {
 
         EurekaInterestClient interestClient = new FullFetchInterestClient(registry, channelFactory);
 
-        Module[] modules = {
-                new EurekaReadServerModule(config, registrationClient, interestClient),
-                new AbstractModule() {
+        Module embeddedReadServerModule = Modules.override(new EurekaReadServerModule(config, registrationClient, interestClient))
+                .with(new AbstractModule() {
                     @Override
                     protected void configure() {
                         if (networkRouter != null) {
                             bind(NetworkRouter.class).toInstance(networkRouter);
-                            bind(TcpInterestServer.class).to(EmbeddedTcpInterestServer.class);
+                            bind(TcpInterestServer.class).to(EmbeddedTcpInterestServer.class).in(Scopes.SINGLETON);
```

Comment:
Why in scope is needed here? We have already singleton annotation on the ```EmbeddedTcpInterestServer``` class.